app/src/main/java/org/schabi/newpipe/fragments/list/kiosk/KioskFragment.java

```diff
@@ -173,4 +176,13 @@ public void showEmptyState() {
             setEmptyStateMessage(R.string.no_live_streams);
         }
     }
+
+    @Override
+    protected void onItemSelected(final InfoItem selectedItem) {
+        final Fragment fragment = getFM().findFragmentById(R.id.fragment_player_holder);
+        if (fragment instanceof VideoDetailFragment videoDetailFragment && fragment.isVisible()) {
+            videoDetailFragment.removeLastStackItem();
+        }
+        super.onItemSelected(selectedItem);
+    }
```

Comment:
It seems to work, but I’m not sure if this is the right place to implement this functionality.

It feels a bit hacky to remove the last item when an item is selected, plus the whole “fetch video fragment via the fragment manager” type-cast looks fishy.